config/config_test.go

```diff
@@ -302,3 +384,38 @@ func TestConfigWithSampler(t *testing.T) {
 	require.Equal(t, traceID, sampler.lastTraceID)
 	require.Equal(t, "test", sampler.lastOperation)
 }
+
+func TestNewTracer(t *testing.T) {
+	cfg := &Configuration{ServiceName: "my-service"}
+	_, closer, err := cfg.NewTracer(Metrics(metrics.NullFactory), Logger(log.NullLogger))
+	defer closer.Close()
+
+	require.NoError(t, err)
+}
+
+func TestNewTracerWithoutServiceName(t *testing.T) {
+	cfg := &Configuration{}
+	_, _, err := cfg.NewTracer(Metrics(metrics.NullFactory), Logger(log.NullLogger))
+	require.Error(t, err)
```

Comment:
nit: this test may succeed as false positive. Best to check not only that there was an error, but that the error contained a certain string, e.g. `assert.Contains(t, err.Error(), "pattern")`